src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -1573,6 +1573,7 @@ def getSubscriptionInformation(self, initialTask=None):
                     subInfo[dataset]["NonCustodialGroup"] = solveGroupConflicts(
                         taskSubInfo[dataset]["NonCustodialGroup"],
                         subInfo[dataset]["NonCustodialGroup"])
+                    subInfo[dataset]["Dataset_lifetime"] = solveGroupConflicts(taskSubInfo[dataset]["Dataset_lifetime"], subInfo[dataset]["Dataset_lifetime"])
```

Comment:
the key name here should be `DatasetLifetimeDisk `, as defined in the StdBase spec.